### ISSUE TITLE
Disable color output does not work as expected

### ISSUE THREAD

**Issue by niko2342** [saschagrunert/git-journal#2]
I found a some issues regarding the color output:

1. After setting 'colored_output' to false, the debug output is still formatted with colors.
2. In addition, after setting 'debug_output' to false, the generated changelog contains color informations:

```
# Unreleased (2017-06-11):
- [Fixed] clippy lints^[(B^[[m
- [Fixed] builds with newer Rust versions^[(B^[[m

# 1.4.2 (2017-03-18):
- [Changed] internal error handling to use error-chain crate^[(B^[[m

# 1.4.1 (2017-03-05):
- [Changed] Cargo files to newest version^[(B^[[m
- [Changed] lockfile to current version^[(B^[[m
- [Fixed] kcov build and test^[(B^[[m
```

config:
```
categories = ["Added", "Changed", "Fixed", "Improved", "Removed"]
category_delimiters = ["[", "]"]
colored_output = false 
enable_debug = false 
excluded_commit_tags = []
enable_footers = false
show_commit_hash = false
show_prefix = false
sort_by = "date"
template_prefix = ""
```

**Comment by saschagrunert** [comment]
Okay, sounds like there are some checks missing.

**Comment by saschagrunert** [comment]
Can you please verify if the bug is still present in the latest version 1.6.0?

**Comment by niko2342** [comment]
Hi,

the first bug is fixed. but the second bug still exists. i looked into the code and found the culprit (a missing colored_output test):

```diff
diff --git a/src/parser.rs b/src/parser.rs
index 35069e1..9fb895d 100644
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -523,7 +523,9 @@ impl Print for SummaryElement {
                     write!(t, " ({:.7})", oid)?;
                 }
             }
-            c3(t)?;
+            if config.colored_output {
+                c3(t)?;
+            }
         }
         Ok(Printed::Something)
     }
```

Shall I create a new pull request for this fix or would you like to fix the bug by your own?

**Comment by saschagrunert** [comment]
Alright, please check if it is fixed with 1.6.1

**Comment by niko2342** [comment]
it's fixed! thanks a lot!
